docs/devel/e2e.md

```diff
@@ -32,33 +32,92 @@ Documentation for other releases can be found at
+There are a variety of ways to run e2e tests, but we aim to decrease the number of ways to run e2e tests to a canonical way: `hack/e2e.go`.
 
-### Building the Tests
+You can run an end-to-end test which will bring up a master and nodes, perform some tests, and then tear everything down. Make sure you have followed the getting started steps for your chosen cloud platform (which might involve changing the `KUBERNETES_PROVIDER` environment variable to something other than "gce").
 
-The tests are built into a single binary which can be run against any deployed kubernetes system.  To build the tests, navigate to your source directory and execute:
+To build Kubernetes, up a cluster, run tests, and tear everything down, use:
 
-`$ make all`
+```sh
+go run hack/e2e.go -v --build --up --test --down
+```
+
+If you'd like to just perform one of these steps, here are some examples:
+
+```sh
+# Build binaries for testing
+go run hack/e2e.go --build
+
+# Create a fresh cluster.  Deletes a cluster first, if it exists
+go run hack/e2e.go --up
+
+# Create a fresh cluster at a specific release version.
+go run hack/e2e.go --up --version=0.7.0
+
+# Test if a cluster is up.
+go run hack/e2e.go --isup
+
+# Push code to an existing cluster
+go run hack/e2e.go --push
+
+# Push to an existing cluster, or bring up a cluster if it's down.
+go run hack/e2e.go --pushup
+
+# Run all tests
+go run hack/e2e.go --test
 
-The output for the end-2-end tests will be a single binary called `e2e.test` under the default output directory, which is typically `_output/local/bin/linux/amd64/`.  Within the repository there are scripts that are provided under the `./hack` directory that are helpful for automation, but may not apply for a local development purposes.  Instead, we recommend familiarizing yourself with the executable options.  To obtain the full list of options, run the following:
+# Run tests matching the regex "\[Conformance\]" (the conformance tests)
+go run hack/e2e.go -v -test --test_args="--ginkgo.focus=\[Conformance\]"
 
-`$ ./e2e.test --help`
+# Conversely, exclude tests that match the regex "Pods.*env"
+go run hack/e2e.go -v -test --test_args="--ginkgo.focus=Pods.*env"
 
-### Running the Tests
+# Flags can be combined, and their actions will take place in this order:
+# --build, --push|--up|--pushup, --test|--tests=..., --down
+#
+# -v (verbose) can be added if you want streaming output instead of only
+# seeing the output of failed commands.
+#
+# e.g.:
+go run hack/e2e.go -v --build --pushup --test --down
 
-For the purposes of brevity, we will look at a subset of the options, which are listed below:
+# -ctl can be used to quickly call kubectl against your e2e cluster. Useful for
+# cleaning up after a failed test or viewing logs. Use -v to avoid suppressing
+# kubectl output.
+go run hack/e2e.go -v -ctl='get events'
+go run hack/e2e.go -v -ctl='delete pod foobar'
```

Comment:
Oh; if you have a system installation, `kubectl` will route to that, not the locally-built one.  This is a shortcut to `./cluster/kubectl.sh` IIUC.  Why, though, I couldn't tell you.  I'm just munging docs.  :stuck_out_tongue: